src/vstart.sh

```diff
@@ -757,6 +769,13 @@ EOF
 		ln -s $kstore_path $CEPH_DEV_DIR/osd$osd
 	    else
 		mkdir -p $CEPH_DEV_DIR/osd$osd
+		if [ -n "${bluestore_dev[$osd]}" ]; then
+		    dd if=/dev/zero of=${bluestore_dev[$osd]} bs=1M count=1
+		    ln -s ${bluestore_dev[$osd]} $CEPH_DEV_DIR/osd$osd/block
```

Comment:
Add some quotes here too please for the variables.

Reply:
Meh, ok.

Reply:
Because `${bluestore_dev{$osd}}` may expand, even if unlikely. For example:

````
pdonnell@icewind ~$ args "${FOO[1]}"
00: "foo bar"
pdonnell@icewind ~$ args ${FOO[1]}
00: "foo"
01: "bar"
````

Using this helper function:

````
function args {
    local args=("$@")
    for ((i = 0; i < "${#args[@]}"; i++)); do
        printf '%02d: "%s"\n' "$i" "${args[${i}]}"
    done
}
````

Reply:
Seems unnecessary here. We're dealing with command-line arguments so we know there shouldn't be embedded spaces in here. If you pass in a quoted argument that has them, then I think you get to keep both pieces when it breaks.